### ISSUE TITLE
Touch upon args size in sd_bus_message_read() docs

### ISSUE THREAD

**Issue by hadess** [systemd/systemd#14643]
Touch upon arguments alignment sd_bus_message_read()

The code:
```
sd_bus_message_read (m, "sb", &result, &done));
```
will not work the same way if `done` is declared as `bool` or if it is declared as `uint64_t`.

In my test case, which I unfortunately haven't been able to replicate, using a `bool` means that `result` address gets truncated.

```
#3  0x00007fffe9879495 in verify_result (m=0x555555727670, userdata=0x55555572f9f0, ret_error=<optimized out>) at /home/hadess/Projects/jhbuild/fprintd/pam/pam_fprintd.c:206
206		printf ("result: %s\n", result);
(gdb) p result
$1 = 0x555555000000 <error: Cannot access memory at address 0x555555000000>
```

**Comment by poettering** [comment]
needs to be an "int".

See table on:

https://www.freedesktop.org/software/systemd/man/sd_bus_message_read.html

I mean, our documentation of sd-bus is unfortunately very much incomplete, but this specific point is actually documented.

But I guess it wouldn't hurt underlining this specifically, since the "bool" vs. "int" stuff is a trap too easy to fall into. will prep patch

**Comment by poettering** [comment]
Fix waiting in #14685
